src/main/scala/mimir/statistics/DetectSeries.scala

```diff
@@ -44,13 +44,47 @@ class DetectSeries(db: Database) {
         case _ => None
       }
     }
-//    Not working! Typechecker.schemaOf is not detecting type for csv loaded files( detected as 'Any')
-//    val seqCol = Typechecker.schemaOf(oper) 
     val seqCol = db.bestGuessSchema(oper)
     
-    val seriesColTemp = seqCol.flatMap( tup => getType(tup._1, tup._2))
+    val seriesColDate = seqCol.flatMap( tup => getType(tup._1, tup._2))
+
+    var series = seriesColDate.toList
+    
+    val seriesColNumIdx: Seq[Int] = seqCol.zipWithIndex.filter(tup => Seq(TInt(),TFloat()).contains(tup._1._2)).map(x => x._2).toSeq
     
-    seriesColTemp.toList
+    if(!seriesColNumIdx.isEmpty){
+      db.query(oper){result =>
+//        var prevVal = new ArrayBuffer[PrimitiveValue](seriesColNumIdx.size)
+        val initRow = result.next
+        var prevVal = seriesColNumIdx.map(initRow.tuple(_)).toArray
+        var countInc = new Array[Double](seriesColNumIdx.size)
+        var countDec = new Array[Double](seriesColNumIdx.size)
+        var totalTup = 1;
+        result.foreach(row => 
+          { totalTup = totalTup+1;
+            seriesColNumIdx.zipWithIndex.foreach(x => 
+              if(db.interpreter.evalBool(prevVal(x._2).lt(row.tuple(x._1))))
+                { prevVal(x._2)=row.tuple(x._1); countInc(x._2) = countInc(x._2)+1.0}
+              else if(db.interpreter.evalBool(prevVal(x._2).eq(row.tuple(x._1)))){
+                prevVal(x._2)=row.tuple(x._1); countInc(x._2) = countInc(x._2)+1.0; countDec(x._2) = countDec(x._2)+1.0;
+              }
+              else
+                { 
+                  if(!db.interpreter.evalBool(row.tuple(x._1).isNull.or(prevVal(x._2).isNull)))
+                  {countDec(x._2) = countDec(x._2)+1.0}
+                  prevVal(x._2)=row.tuple(x._1);
+                 })
```

Comment:
In general, queries produce results in arbitrary order – You can't rely on them showing up in the right 'sorted' order.
1. You will need to sort the data on each permitted column.  Use [query.sort](https://github.com/UBOdin/mimir/blob/master/src/main/scala/mimir/algebra/OperatorConstructors.scala#L113) to get a sorted version of the query.
2. Since the data will already be sorted, the important thing is seeing if that sort order makes sense.  I would suggest first computing a sequence of `curr - prev` values, for example with [Iterator](http://www.scala-lang.org/api/2.11.8/#scala.collection.Iterator)'s `sliding` window function and then seeing if these values are mostly self-consistent (e.g., `Math.abs(stddev/avg) < 0.1`).